script/tasks/diamondSyncWhitelist.sh

```
@@ -617,52 +1007,161 @@ function diamondSyncWhitelist {
+        
+        # Validate that we have arrays to process
+        if [[ -z "$REMOVE_CONTRACTS_ARRAY" || -z "$REMOVE_SELECTORS_ARRAY" ]]; then
+          printf '\033[0;31m%s\033[0m\n' "❌ [$NETWORK] Cannot construct calldata: empty arrays"
+          echoSyncDebug "REMOVE_CONTRACTS_ARRAY: '$REMOVE_CONTRACTS_ARRAY'"
+          echoSyncDebug "REMOVE_SELECTORS_ARRAY: '$REMOVE_SELECTORS_ARRAY'"
+          return
+        fi
+        
+        local CONTRACTS_FOR_CALLDATA="$REMOVE_CONTRACTS_ARRAY"
+        if isTronNetwork "$NETWORK"; then
+          local TRON_ENV=$(getTronEnv "$NETWORK")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)